nixos/tests/ollama.nix

```diff
@@ -32,6 +32,7 @@ in
     };
 
     cuda = { ... }: {
+      nixpkgs.config.allowUnfree = true;
```

Comment:
This will break setups that pass pkgs to lib.nixosSystem. There is an assert in that module that fails in this case. I'd suggest removing this line.

Reply:
I don't understand why this is a problem. Can you explain in more detail how `lib.nixosSystem` would fail because a test vm enabled `allowUnfree`? What assert in what module?

Also, why pass `pkgs` to `lib.nixosSystem`? I don't see what the purpose of that is, or what that might accomplish.

This line was added to allow ofborg to directly run ollama's nixos test (ie `@ofborg test ollama`) since it previously failed to evaluate it, complaining about the unfree cuda dependencies. Do you know any way to solve this problem without causing errors when passing `pkgs` to `lib.nixosSystem`?

Reply:
> This will break setups that pass pkgs to lib.nixosSystem.

This is a nixos test, not nixos module, so it itself is responsible for calling nixosSystem?

> This line was added to allow ofborg to directly run ollama's nixos test

I'm not sure whether we want _the_ ofborg to test unfree things. I think so far it hasn't (except by accident). That said, we always wanted _an_ ofborg to do that

Reply:
> I'm not sure whether we want _the_ ofborg to test unfree things.

Why not?

> we always wanted _an_ ofborg to do that

What's wrong with the current ofborg?

Reply:
I overlook the fact that this file is actually a test :facepalm: 

But in the main module that would be a problem. In my config, for example, I instantiate a outer pkgs then pass to all my machines and home-manager. Easier to add stuff that cascades into the machines.

Reply:
> I overlook the fact that this file is actually a test 🤦

I would never add `nixpkgs.config.allowUnfree = true;` to an actual nixos module. I can definitely see why you would be against that.

> pass pkgs to lib.nixosSystem

So were you not referring here to `nixpkgs.lib.nixosSystem (import nixpkgs {})`?

Reply:
> What's wrong with the current ofborg?

It's just the risk aversity regarding the official infra and the unfree. Another issue might be, ofborg won't be able to substitute cuda stuff using the official cache.

That said, Ofborg's build outputs aren't redistributed, and we could in principle make Ofborg use the nix-community cache, so I'm not sure if there's any policy or decision regarding Ofborg specifically. And eitherways if there is one, now's a good time to reconsider.

But yeah I think this change involves more parties that must be made aware of the change and consulted with...

Reply:
> ofborg won't be able to substitute cuda stuff using the official cache.

Again, why not? Not all unfree licenses restrict redistribution of binaries or other build artifacts. Does cuda?

> But yeah I think this change involves more parties that must be made aware of the change and consulted with...

Yeah, I agree. I don't actually know who any of those people are, though. Do you?

Reply:
> Again, why not? Not all unfree licenses restrict redistribution of binaries or other build artifacts. Does cuda?

Ofborg won't be able to substitute because the official hydra doesn't build unfree, and the official hydra doesn't build unfree because last time the proposal met a quite a bit of opposition (e.g. https://discourse.nixos.org/t/petition-to-build-and-cache-unfree-packages-on-cache-nixos-org/, but there were discussions before and after). Not that this is set in stone, but also this is probably OK if we can rely on the nix-community hydra instead. Ofborg and CI are yet to be figured out

> Yeah, I agree. I don't actually know who any of those people are, though. Do you?

One could start by posting a proposal on discourse, arguing that there's no legal threat (is there?) in just having Ofborg test the unfree stuff without redistribution (is there?) so why don't we just enable it.

If that meets opposition, we look for an opportunity to run (or co-host) a separate ofborg-like CI instance that could be integrated with github...

Reply:
Also cf https://github.com/NixOS/ofborg/pull/644